src/crimson/os/seastore/cached_extent.h

```diff
@@ -1023,6 +1085,17 @@ class LogicalCachedExtent : public CachedExtent {
 private:
   laddr_t laddr = L_ADDR_NULL;
   LBAPinRef pin;
+
+  template <
+    size_t CAPACITY,
+    typename NODE_KEY,
+    typename NODE_KEY_LE,
+    typename VAL,
+    typename VAL_LE,
+    size_t node_size,
+    typename node_type_t,
+    bool has_children>
+  friend struct FixedKVLeafNode;
```

Comment:
Adding friends is not helpful to maintain well abstractions and keep the code structured. It's always preferred to adjust the interfaces instead.

Adjusting interfaces also helps to better understand how the exisiting class interacts with the proposed change.

Reply:
I'll remove all other fixed-kv-tree related friends, but leaving one that make `FixedKVBtree` a friend to `CachedExtent`. This is because `FixedKVBtree` needs to set extents' prior_instance when rewriting them; and adding an interface for setting prior_instance would give other components the ability to change the field, which, I think, we should avoid. What do you think?

Reply:
> This is because FixedKVBtree needs to set extents' prior_instance when rewriting them; and adding an interface for setting prior_instance would also give other components the ability to change the field

How about introducing a `FixedKVNode::get_rewrite_node(t) -> n_fixed_kv_extent`, so `set_prior_instance()` can be protected?